app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -4,8 +4,36 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
   require_authentication :all, scopes: [:project]
   schema_type :json_schema
 
+  def update_project_settings
+    binding.pry
+    project = Project.find params[:project_id]
+    unless required_params
```

Comment:
If you wire up the params_for via the json schema validator you get this for free.